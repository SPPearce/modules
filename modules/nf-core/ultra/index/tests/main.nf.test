nextflow_process {

    name "Test Process ULTRA_INDEX"
    config "./nextflow.config"
    script "../main.nf"
    process "ULTRA_INDEX"

    tag "modules"
    tag "modules_nfcore"
    tag "ultra"
    tag "ultra/index"
    tag "gnu"
    tag "gnu/sort"

    test("test_ultra_index") {

        setup {
            run("GNU_SORT") {
                script "../../../gnu/sort/main.nf"

                process {
                    """
                    input[0] = [
                        [id:'test'],
                        file(params.modules_testdata_base_path + 'genomics/homo_sapiens/genome/genome.gtf',checkIfExists:true)
                        ]
                    """
                }
            }
        }

        when {
            params {
                sort_args   = '-k1,1 -k4,4n'
                module_args = '--disable_infer'
            }
            process {
                """
                input[0] = file(params.modules_testdata_base_path + 'genomics/homo_sapiens/genome/genome.fasta',checkIfExists:true)
                input[1] = GNU_SORT.out.sorted.map{it[1]}
                """
            }
        }
        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }
}