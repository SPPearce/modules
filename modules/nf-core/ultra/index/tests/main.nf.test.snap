{
    "test_ultra_index": {
        "content": [
            {
                "0": [
                    [
                        [
                            "all_splice_pairs_annotations.pickle:md5,ec60c02903edc30a3f10c196c425fb16",
                            "all_splice_sites_annotations.pickle:md5,5a7d640f4e707acb37966542c132455a",
                            "chr_to_id.pickle:md5,b8aa0a986ec685f5f7e8fad651ff1b90",
                            "exon_choordinates_to_id.pickle:md5,4e978d727a90e2b225c5ff4bc6107578",
                            "flank_choordinates.pickle:md5,309e3de05b9d781e01810ea442ebf2ef",
                            "gene_to_small_segments.pickle:md5,37a25f3ba84edac017ecb9e946081696",
                            "id_to_chr.pickle:md5,020263c6d9f399670871ca187c1d203f",
                            "max_intron_chr.pickle:md5,f568beb2f859d8e9b1f2993048e4fcd2",
                            "parts_to_segments.pickle:md5,56950e119f074e4b4577dc3cbf00ea76",
                            "ref_exon_sequences.pickle:md5,860192c5711d1b41bda2d3677cf310f5",
                            "ref_flank_sequences.pickle:md5,9b81749e32c4e4f742590218e8854313",
                            "ref_part_sequences.pickle:md5,3354b0d8f8f1c9db1a98d70e470826ff",
                            "ref_segment_sequences.pickle:md5,fe89d6588433b848a2055395765032a6",
                            "refs_id_lengths.pickle:md5,102873a7c330dc1ca562a699dec2d5f2",
                            "refs_lengths.pickle:md5,93012591af522a2f6a6c1620f23aa065",
                            "segment_id_to_choordinates.pickle:md5,59689a3f3e8bebb1938d576932f266f8",
                            "segment_to_gene.pickle:md5,b27dbc0e09354a637c453a9130aae260",
                            "segment_to_ref.pickle:md5,628eb8bd24cda6f7dd6a2de7f591ffda",
                            "splices_to_transcripts.pickle:md5,4b7d114268be85793fa0b2de64c8adbe",
                            "transcripts_to_splices.pickle:md5,3e307206909ed029a69b65f6afca02d0"
                        ],
                        "database.db:md5,f9b780bf5a0a5d23256afb99df7137d9"
                    ]
                ],
                "1": [
                    "versions.yml:md5,45fdc5637701f0c91e005244cc09a199"
                ],
                "index": [
                    [
                        [
                            "all_splice_pairs_annotations.pickle:md5,ec60c02903edc30a3f10c196c425fb16",
                            "all_splice_sites_annotations.pickle:md5,5a7d640f4e707acb37966542c132455a",
                            "chr_to_id.pickle:md5,b8aa0a986ec685f5f7e8fad651ff1b90",
                            "exon_choordinates_to_id.pickle:md5,4e978d727a90e2b225c5ff4bc6107578",
                            "flank_choordinates.pickle:md5,309e3de05b9d781e01810ea442ebf2ef",
                            "gene_to_small_segments.pickle:md5,37a25f3ba84edac017ecb9e946081696",
                            "id_to_chr.pickle:md5,020263c6d9f399670871ca187c1d203f",
                            "max_intron_chr.pickle:md5,f568beb2f859d8e9b1f2993048e4fcd2",
                            "parts_to_segments.pickle:md5,56950e119f074e4b4577dc3cbf00ea76",
                            "ref_exon_sequences.pickle:md5,860192c5711d1b41bda2d3677cf310f5",
                            "ref_flank_sequences.pickle:md5,9b81749e32c4e4f742590218e8854313",
                            "ref_part_sequences.pickle:md5,3354b0d8f8f1c9db1a98d70e470826ff",
                            "ref_segment_sequences.pickle:md5,fe89d6588433b848a2055395765032a6",
                            "refs_id_lengths.pickle:md5,102873a7c330dc1ca562a699dec2d5f2",
                            "refs_lengths.pickle:md5,93012591af522a2f6a6c1620f23aa065",
                            "segment_id_to_choordinates.pickle:md5,59689a3f3e8bebb1938d576932f266f8",
                            "segment_to_gene.pickle:md5,b27dbc0e09354a637c453a9130aae260",
                            "segment_to_ref.pickle:md5,628eb8bd24cda6f7dd6a2de7f591ffda",
                            "splices_to_transcripts.pickle:md5,4b7d114268be85793fa0b2de64c8adbe",
                            "transcripts_to_splices.pickle:md5,3e307206909ed029a69b65f6afca02d0"
                        ],
                        "database.db:md5,f9b780bf5a0a5d23256afb99df7137d9"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,45fdc5637701f0c91e005244cc09a199"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.5"
        },
        "timestamp": "2025-04-01T13:48:22.364563262"
    }
}